--- /data/data/com.termux/files/usr/bin/lxc-setup-cgroups	2025-01-23 20:43:20.079999930 +0800
+++ lxc-setup-cgroups	2025-01-23 20:55:12.027999827 +0800
@@ -3,16 +3,18 @@
 
 export PATH=$PATH:/system/xbin:/system/bin
 
-if ! mountpoint -q /sys/fs/cgroup; then
-	mount -t tmpfs -o mode=755,nodev,noexec,nosuid tmpfs /sys/fs/cgroup
-fi
+mount -t tmpfs -o mode=755 tmpfs /sys/fs/cgroup
 
-for cg in blkio cpu cpuacct cpuset devices freezer memory; do
+for cg in blkio cpu cpuacct cpuset devices freezer memory systemd; do
 	if [ ! -d "/sys/fs/cgroup/${cg}" ]; then
 		mkdir -p "/sys/fs/cgroup/${cg}"
 	fi
 
 	if ! mountpoint -q "/sys/fs/cgroup/${cg}"; then
+		if [ "${cg}" = "systemd" ]; then
+			mount -t cgroup -o none,name=systemd cgroup /sys/fs/cgroup/systemd
+			continue
+		fi
 		mount -t cgroup -o "${cg}" cgroup "/sys/fs/cgroup/${cg}" || true
 	fi
 done
